### ISSUE TITLE
Broker hangs when mosquitto.db.new is created

### ISSUE THREAD

**Issue by catan85** [eclipse/mosquitto#172]
I'm using mosquitto in a raspberry with persistence activated. 
All the messages are forwarded to another broker. 
When the second broker is offline messages are saved on mosquitto.db file.
Suddenly it happend that a file called mosquitto.db.new is created. During this process the broker stops accepting messages untill the process is terminated and mosquitto.db.new file disappears.
Is it a normal behaviour?
What can I do to prevent it? Maybe use a dual core processor? or use a board with more ram?

**Comment by catan85** [comment]
I already tried with a powerfull pc but nothing changed. When this file is created the broker stops responding to the subscriptions.

**Comment by jpmens** [comment]
What OS/version and which Mosquitto version are you running?
What amount of messages / time is the broker receiving and how large (approximately) are the payloads?

I've never seen this happening on Raspi (neither 1, 2, nor 3) and we run quite a number of bridges.

**Comment by karlp** [comment]
yeah, I see this, it's only an issue if you have _big_ persistence dbs.  (we have some persistence dbs that are ~1gig) There's some prototype code in the sqlite branch that avoids this quite nicely :)  a raspi with a dirt slow sd card makes the problem substantially more obvious.

**Comment by karlp** [comment]
for prevention in the short term:

- faster disk
- turn down queue sizes

**Comment by catan85** [comment]
I was using version 1.4.9
I tried with Debian on a raspi 2, and with Windows 7 on a Inter Core i7 with SSD drive. The behaviour is the same (obviously on the faster pc the downtime was less.. but noticeable anyway).
As @karlp said, the problem comes when the persistence file becomes big (1 Gb in my case). 
Actually I solved the problem by myself by managing the "store and forward" process using an external software writed in python, but I would prefer to avoid any external code..
Where can I find that sqlite branch that you mention @karlp?
Sorry for my english.. :)
